### PR TITLE
Fix check decorator and ctx mgr read of binary evidence

### PR DESCRIPTION
**Pull request by alfinkel**
- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](../blob/master/DCO1.1.txt)

## What

- Fix check decorator and ctx mgr read of binary evidence
- Added f/c's examples in `demo`
- Cleaned up Design Principles format
- Fixed Quick Start example to create virtual environment

## Why

So checks can process binary evidence

## How

- If evidence is binary content then open the file in `b`inary mode using the `Locker`.
- Added f/c's examples in `demo`
- Cleaned up Design Principles format
- Fixed Quick Start example to create virtual environment

## Test

- Added f/c's examples in `demo` and they run as expected
- Previous functionality not compromised, running tests locally

## Context

#95

### CODE REVIEW

**Comment by alfinkel** [comment]
@drsm79 if you're around?

File: compliance/locker.py [hunk 1]
```diff
@@ -787,9 +791,10 @@ def _evidence_abandoned(self, metadata=None, threshold=None):
             return True
         return False
 
-    def _get_file_content(self, file_path, file_dt=None):
+    def _get_file_content(self, file_path, file_dt=None, binary_content=False):
         if not file_dt:
-            return open(os.path.join(self.local_path, file_path)).read()
+            mode = 'rb' if binary_content else 'r'
+            return open(os.path.join(self.local_path, file_path), mode).read()
```
Comment (drsm79) [inline compliance/locker.py:797]: Pathlib! ;)

Comment (alfinkel) [reply compliance/locker.py:797]: some day!

File: demo/demo_examples/fetchers/fetch_auditree_logo.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+from compliance.evidence import store_raw_evidence
+from compliance.fetch import ComplianceFetcher
+
+class ImageFetcher(ComplianceFetcher):
+    """Fetch the Auditree logo image and store as evidence."""
+
+    @store_raw_evidence('images/auditree_logo.png')
+    def fetch_auditree_logo(self):
+        """Fetch the Auditree logo."""
+        return open('at-logo.png', 'rb').read()
```
Comment (drsm79) [inline demo/demo_examples/fetchers/fetch_auditree_logo.py:25]: Pathlib?

Comment (alfinkel) [reply demo/demo_examples/fetchers/fetch_auditree_logo.py:25]: some day?

**Review by drsm79 — APPROVED** [review]

**Pull request merged**